### PR TITLE
docs: patterns

### LINKED ISSUE #967: [docs] announcement pattern spacing
Announcement pattern spacing section says to use Patternfly 4 spacers.  Assuming we would like to update this to say @rhds/tokens instead

image

### PR DESCRIPTION
**Pull request by bennypowers**
Closes #967 

## What I did

1. migrate patterns from njk to markdown
2. move patterns page assets into patterns dir
3. make `srcAbsolute` flag on example shortcode derive from src url
4. remove references to patternfly spacers

### CODE REVIEW

**Comment by nikkimk** [comment]
I'll grab **Fliter**

**Comment by brianferry** [comment]
updated the promo and search bar docs, but alert needs to be added via a shortcode that I haven't yet created

**Comment by bennypowers** [comment]
> updated the promo and search bar docs, but alert needs to be added via a shortcode that I haven't yet created

we already have one. see [0271528](https://github.com/RedHat-UX/red-hat-design-system/pull/1090/commits/02715286127d198101c5d7e930bc038d3950c882)

File: docs/patterns/link-with-icon/index.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+           alt="Link with icon alignment",
+           src="./link-with-icon-alignment-2.svg" %}
+
+## Interaction states
+
+The link has the same interaction states as a Link whereas the icon doesn't have any interaction states.
+
+## Spacing
+
+Link with icon uses [PatternFly 4 spacers](https://www.patternfly.org/v4/guidelines/spacers) to define spacing values between elements.
```
Comment (bennypowers) [inline docs/patterns/link-with-icon/index.md:85]: change to tokens links. see announcement

File: docs/patterns/disclosure/index.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+![Example of a disclosure on desktop](./disclosure-responsive-1.svg)
+
+### Tablet
+
+![Example of a disclosure on tablet](./disclosure-responsive-2.svg)
+
+### Mobile
+
+![Example of a disclosure on mobile](./disclosure-responsive-3.svg)
```
Comment (zeroedin) [inline docs/patterns/disclosure/index.md:210]: We should use the example shortcode here to ensure proper image sizing.

File: docs/patterns/sticky-card/index.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+### Body
+
+To keep the vertical height short, the body section can only include a small amount of text or a small image thumbnail. If a card title and a headline are enough to explain what the content is, the body section can be hidden.
+
+### Footer
+
+The footer section can include normal links or a call to action. A sticky card footer is always required and shouldn't be hidden.
+
+{% example palette="lightest",
```
Comment (zeroedin) [inline docs/patterns/sticky-card/index.md:58]: Scaling of image is off here as well.

File: docs/patterns/sticky-card/index.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+{% example palette="medium",
+        alt="Sticky card on right side",
+        src="./sticky-card-layout-right.svg" %}
+
+### Content
+
+A sticky card has limited vertical height, so keep content short and only include essential information. A sticky card can include a card title, a headline, text, and a call to action, but not all of these elements need to be included at the same time. Move any extra content to other parts of the page if possible.
+
+{% example palette="lightest",
```
Comment (zeroedin) [inline docs/patterns/sticky-card/index.md:78]: here as well

**Comment by bennypowers** [comment]
At this point, whatever's left over, let's spin off into new issues for bsaur

File: docs/patterns/disclosure/index.md (hunk 2 above)
Comment (zeroedin) [reply docs/patterns/disclosure/index.md]: Spun out for its own issue, issue with example shortcode and svgs that and double scaled

File: docs/patterns/sticky-card/index.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+          alt="Sticky card",
+          src="./sticky-card.svg" %}
+
+{% repoStatus %}
+
+## Style
+
+A sticky card acts as a small container for a limited amount of content.
+
+{% example palette="lightest",
```
[the remark this answers is not in the document]
Comment (zeroedin) [reply docs/patterns/sticky-card/index.md:22]: Spun out for its own issue, issue with example shortcode and svgs that and double scaled

File: docs/patterns/sticky-card/index.md (hunk 4 above)
Comment (zeroedin) [reply docs/patterns/sticky-card/index.md:78]: Spun out for its own issue, issue with example shortcode and svgs that and double scaled

**Review by zeroedin — APPROVED** [review]
Spun off remaining issues to their own issue for a fast follow.  LGTM

**Pull request merged**